shell/app.cc

```diff
@@ -35,9 +37,16 @@ App::App(const std::vector<Configuration::Config>& configs)
     view->Initialize();
     m_views.emplace_back(std::move(view));
     index++;
+
+    if (WaylandWindow::get_window_type(cfg.view.window_type) ==
```

Comment:
I think this is in conflict with my recent change.  I make BG the default window type.  The reasoning is so that one can run cli options not including json config, as json config was only way to set window type.

Reply:
Yeah, pretty much. 

> The reasoning is so that one can run cli options not including json config

Not really sure how to reconcile that. To me this sounds like a reason to keep it as normal, as the default, while for background/panel being enabled if either a) explicitly specified over config file, or b) passed over as cmd line. 

This way:

- for non-AGL platforms, we resort to using normal role (and the agl calls are stubs as it won't find the protocol interface being advertised)
- on AGL platforms where the flutter platform is not the default shell client, it can run flutter as regular/normal applications
- on AGL platforms where the flutter platforms is the shell client, you explicitly state it (and avoid a potential conflict with other running flutter apps). 

At some point in time, I'll update the protocol interface such that the client can still bind to the protocol interface and determine if there's another shell client running. With that in place, the client can infer what do do next (it lost the race, there's another shell client running, or not). Until that happens, adding an explicit command line argument or using the config file seems to be the easiest to go forward.

Reply:
I think the way to reconcile is pass widow type as cli option.  Then revert default back to `normal`.  I'll create a PR for this.  I just wanted to unblock the need to pass --j=

Reply:
Well then I guess with this change we break again functionality of ivi-homescreen as our applications will need to pass that argument.